File: onyx-database-tests/test/web/delete/DeleteHashIndexEntityTest.java
package web.delete;

import category.WebServerTests;
import com.onyx.exception.OnyxException;
import com.onyx.exception.NoResultsException;
import entities.AllAttributeEntity;
import entities.identifiers.IntegerIdentifierEntity;
import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.experimental.categories.Category;
import web.base.BaseTest;

import java.io.IOException;
import java.util.Date;

/**
 * Created by timothy.osborn on 11/3/14.
 */
@Category({ WebServerTests.class })
public class DeleteHashIndexEntityTest extends BaseTest
{
    @Before
    public void before() throws OnyxException
    {
        initialize();
    }

    @After
    public void after() throws IOException
    {
        shutdown();
    }

    @Test
    public void testDeleteEntity() throws OnyxException {
        AllAttributeEntity entity = new AllAttributeEntity();
        entity.id = "dc5cholqdu5vha5bb6ned8ASDF";
        entity.booleanValue = false;
        entity.doubleValue = 234.3;
        entity.dateValue = new Date();
        entity.stringValue = "Hiya";
        save(entity);

        /*manager.deleteEntity(entity);

        boolean pass = false;
        try {
            manager.find(entity);
        }
        catch (EntityException e)
        {
            if(e instanceof NoResultsException)
            {
                pass = true;
            }
        }
        Assert.assertTrue(pass);*/
    }

    @Test
    public void testDeleteHashAndReSave()
    {
        IntegerIdentifierEntity entity = new IntegerIdentifierEntity();
        entity.identifier = 1;
        save(entity);
        find(entity);
        delete(entity);
        boolean pass = false;
        try {
            manager.find(entity);
        }
        catch (OnyxException e)
        {
            if(e instanceof NoResultsException)
            {
                pass = true;
            }
        }
        Assert.assertTrue(pass);

        entity = new IntegerIdentifierEntity();
        entity.identifier = 1;
        save(entity);

        try {
            manager.find(entity);
        }
        catch (OnyxException e)
        {
            if(e instanceof NoResultsException)
            {
                pass = false;
            }
        }

        Assert.assertTrue(pass);
    }

    @Test
    public void testDeleteRoot()
    {
        IntegerIdentifierEntity entity1 = new IntegerIdentifierEntity();
        entity1.identifier = 1;
        save(entity1);

        IntegerIdentifierEntity entity2 = new IntegerIdentifierEntity();
        entity2.identifier = 2;
        save(entity2);

        IntegerIdentifierEntity entity3 = new IntegerIdentifierEntity();
        entity3.identifier = -1;
        save(entity3);

        IntegerIdentifierEntity entity4 = new IntegerIdentifierEntity();
        entity4.identifier = 8;
        save(entity4);

        IntegerIdentifierEntity entity5 = new IntegerIdentifierEntity();
        entity5.identifier = 6;
        save(entity5);

        IntegerIdentifierEntity entity6 = new IntegerIdentifierEntity();
        entity6.identifier = 10;
        save(entity6);

        delete(entity1);
        find(entity2);
        find(entity3);
        find(entity4);
        find(entity5);
        find(entity6);

        boolean pass = false;
        try {
            manager.find(entity1);
        }
        catch (OnyxException e)
        {
            if(e instanceof NoResultsException)
            {
                pass = true;
            }
        }

        Assert.assertTrue(pass);
    }

    @Test
    public void testDeleteParent()
    {
        IntegerIdentifierEntity entity1 = new IntegerIdentifierEntity();
        entity1.identifier = 1;
        save(entity1);

        IntegerIdentifierEntity entity2 = new IntegerIdentifierEntity();
        entity2.identifier = 2;
        save(entity2);

        IntegerIdentifierEntity entity3 = new IntegerIdentifierEntity();
        entity3.identifier = -1;
        save(entity3);

        IntegerIdentifierEntity entity4 = new IntegerIdentifierEntity();
        entity4.identifier = 8;
        save(entity4);

        IntegerIdentifierEntity entity5 = new IntegerIdentifierEntity();
        entity5.identifier = 6;
        save(entity5);

        IntegerIdentifierEntity entity6 = new IntegerIdentifierEntity();
        entity6.identifier = 10;
        save(entity6);

        delete(entity5);
        find(entity1);
        find(entity2);
        find(entity3);
        find(entity4);
        find(entity6);

        boolean pass = false;
        try {
            manager.find(entity5);
        }
        catch (OnyxException e)
        {
            if(e instanceof NoResultsException)
            {
                pass = true;
            }
        }

        Assert.assertTrue(pass);
    }

    @Test
    public void testDeleteLeft()
    {
        IntegerIdentifierEntity entity1 = new IntegerIdentifierEntity();
        entity1.identifier = 1;
        save(entity1);

        IntegerIdentifierEntity entity2 = new IntegerIdentifierEntity();
        entity2.identifier = 2;
        save(entity2);

        IntegerIdentifierEntity entity3 = new IntegerIdentifierEntity();
        entity3.identifier = 4;
        save(entity3);

        IntegerIdentifierEntity entity7 = new IntegerIdentifierEntity();
        entity7.identifier = 3;
        save(entity7);

        IntegerIdentifierEntity entity4 = new IntegerIdentifierEntity();
        entity4.identifier = 8;
        save(entity4);

        IntegerIdentifierEntity entity5 = new IntegerIdentifierEntity();
        entity5.identifier = 6;
        save(entity5);

        IntegerIdentifierEntity entity6 = new IntegerIdentifierEntity();
        entity6.identifier = 10;
        save(entity6);


        delete(entity3);
        find(entity1);
        find(entity2);
        find(entity4);
        find(entity5);
        find(entity6);
        find(entity7);

        boolean pass = false;
        try {
            manager.find(entity3);
        }
        catch (OnyxException e)
        {
            if(e instanceof NoResultsException)
            {
                pass = true;
            }
        }

        Assert.assertTrue(pass);
    }

    @Test
    public void testDeleteRight()
    {
        IntegerIdentifierEntity entity1 = new IntegerIdentifierEntity();
        entity1.identifier = 1;
        save(entity1);

        IntegerIdentifierEntity entity2 = new IntegerIdentifierEntity();
        entity2.identifier = 2;
        save(entity2);


        IntegerIdentifierEntity entity4 = new IntegerIdentifierEntity();
        entity4.identifier = 8;
        save(entity4);

        IntegerIdentifierEntity entity5 = new IntegerIdentifierEntity();
        entity5.identifier = 6;
        save(entity5);

        IntegerIdentifierEntity entity6 = new IntegerIdentifierEntity();
        entity6.identifier = 10;
        save(entity6);

        IntegerIdentifierEntity entity8 = new IntegerIdentifierEntity();
        entity8.identifier = 11;
        save(entity8);


        IntegerIdentifierEntity entity3 = new IntegerIdentifierEntity();
        entity3.identifier = 4;
        save(entity3);

        IntegerIdentifierEntity entity7 = new IntegerIdentifierEntity();
        entity7.identifier = 3;
        save(entity7);

        delete(entity6);
        find(entity1);
        find(entity2);
        find(entity3);
        find(entity4);
        find(entity7);
        find(entity8);

        boolean pass = false;
        try {
            manager.find(entity6);
        }
        catch (OnyxException e)
        {
            if(e instanceof NoResultsException)
            {
                pass = true;
            }
        }

        Assert.assertTrue(pass);
    }

}

File: onyx-database-tests/test/memory/delete/DeleteHashIndexEntityTest.kt
package memory.delete

import category.InMemoryDatabaseTests
import com.onyx.exception.OnyxException
import com.onyx.exception.NoResultsException
import com.onyx.persistence.IManagedEntity
import entities.AllAttributeEntity
import entities.identifiers.IntegerIdentifierEntity
import org.junit.After
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.experimental.categories.Category

import java.io.IOException
import java.util.Date

/**
 * Created by timothy.osborn on 11/3/14.
 */
@Category(InMemoryDatabaseTests::class)
class DeleteHashIndexEntityTest : memory.base.BaseTest() {
    @Before
    @Throws(OnyxException::class)
    fun before() {
        initialize()
    }

    @After
    @Throws(IOException::class)
    fun after() {
        shutdown()
    }

    @Test
    @Throws(OnyxException::class)
    fun testDeleteEntity() {
        val entity = AllAttributeEntity()
        entity.id = "dc5cholqdu5vha5bb6ned8ASDF"
        entity.booleanValue = false
        entity.doubleValue = 234.3
        entity.dateValue = Date()
        entity.stringValue = "Hiya"
        save(entity)

        /*manager.deleteEntity(entity);

        boolean pass = false;
        try {
            manager.find(entity);
        }
        catch (EntityException e)
        {
            if(e instanceof NoResultsException)
            {
                pass = true;
            }
        }
        Assert.assertTrue(pass);*/
    }

    @Test
    fun testDeleteHashAndReSave() {
        var entity = IntegerIdentifierEntity()
        entity.identifier = 1
        save(entity)
        find(entity)
        delete(entity)
        var pass = false
        try {
            manager.find<IManagedEntity>(entity)
        } catch (e: OnyxException) {
            if (e is NoResultsException) {
                pass = true
            }
        }

        Assert.assertTrue(pass)

        entity = IntegerIdentifierEntity()
        entity.identifier = 1
        save(entity)

        try {
            manager.find<IManagedEntity>(entity)
        } catch (e: OnyxException) {
            if (e is NoResultsException) {
                pass = false
            }
        }

        Assert.assertTrue(pass)
    }

    @Test
    fun testDeleteRoot() {
        val entity1 = IntegerIdentifierEntity()
        entity1.identifier = 1
        save(entity1)

        val entity2 = IntegerIdentifierEntity()
        entity2.identifier = 2
        save(entity2)

        val entity3 = IntegerIdentifierEntity()
        entity3.identifier = -1
        save(entity3)

        val entity4 = IntegerIdentifierEntity()
        entity4.identifier = 8
        save(entity4)

        val entity5 = IntegerIdentifierEntity()
        entity5.identifier = 6
        save(entity5)

        val entity6 = IntegerIdentifierEntity()
        entity6.identifier = 10
        save(entity6)

        delete(entity1)
        find(entity2)
        find(entity3)
        find(entity4)
        find(entity5)
        find(entity6)

        var pass = false
        try {
            manager.find<IManagedEntity>(entity1)
        } catch (e: OnyxException) {
            if (e is NoResultsException) {
                pass = true
            }
        }

        Assert.assertTrue(pass)
    }

    @Test
    fun testDeleteParent() {
        val entity1 = IntegerIdentifierEntity()
        entity1.identifier = 1
        save(entity1)

        val entity2 = IntegerIdentifierEntity()
        entity2.identifier = 2
        save(entity2)

        val entity3 = IntegerIdentifierEntity()
        entity3.identifier = -1
        save(entity3)

        val entity4 = IntegerIdentifierEntity()
        entity4.identifier = 8
        save(entity4)

        val entity5 = IntegerIdentifierEntity()
        entity5.identifier = 6
        save(entity5)

        val entity6 = IntegerIdentifierEntity()
        entity6.identifier = 10
        save(entity6)

        delete(entity5)
        find(entity1)
        find(entity2)
        find(entity3)
        find(entity4)
        find(entity6)

        var pass = false
        try {
            manager.find<IManagedEntity>(entity5)
        } catch (e: OnyxException) {
            if (e is NoResultsException) {
                pass = true
            }
        }

        Assert.assertTrue(pass)
    }

    @Test
    fun testDeleteLeft() {
        val entity1 = IntegerIdentifierEntity()
        entity1.identifier = 1
        save(entity1)

        val entity2 = IntegerIdentifierEntity()
        entity2.identifier = 2
        save(entity2)

        val entity3 = IntegerIdentifierEntity()
        entity3.identifier = 4
        save(entity3)

        val entity7 = IntegerIdentifierEntity()
        entity7.identifier = 3
        save(entity7)

        val entity4 = IntegerIdentifierEntity()
        entity4.identifier = 8
        save(entity4)

        val entity5 = IntegerIdentifierEntity()
        entity5.identifier = 6
        save(entity5)

        val entity6 = IntegerIdentifierEntity()
        entity6.identifier = 10
        save(entity6)


        delete(entity3)
        find(entity1)
        find(entity2)
        find(entity4)
        find(entity5)
        find(entity6)
        find(entity7)

        var pass = false
        try {
            manager.find<IManagedEntity>(entity3)
        } catch (e: OnyxException) {
            if (e is NoResultsException) {
                pass = true
            }
        }

        Assert.assertTrue(pass)
    }

    @Test
    fun testDeleteRight() {
        val entity1 = IntegerIdentifierEntity()
        entity1.identifier = 1
        save(entity1)

        val entity2 = IntegerIdentifierEntity()
        entity2.identifier = 2
        save(entity2)


        val entity4 = IntegerIdentifierEntity()
        entity4.identifier = 8
        save(entity4)

        val entity5 = IntegerIdentifierEntity()
        entity5.identifier = 6
        save(entity5)

        val entity6 = IntegerIdentifierEntity()
        entity6.identifier = 10
        save(entity6)

        val entity8 = IntegerIdentifierEntity()
        entity8.identifier = 11
        save(entity8)


        val entity3 = IntegerIdentifierEntity()
        entity3.identifier = 4
        save(entity3)

        val entity7 = IntegerIdentifierEntity()
        entity7.identifier = 3
        save(entity7)

        delete(entity6)
        find(entity1)
        find(entity2)
        find(entity3)
        find(entity4)
        find(entity7)
        find(entity8)

        var pass = false
        try {
            manager.find<IManagedEntity>(entity6)
        } catch (e: OnyxException) {
            if (e is NoResultsException) {
                pass = true
            }
        }

        Assert.assertTrue(pass)
    }

}
